forem/forem
# Fix bullet issue with includes commentable

## What type of PR is this? (check all applicable)


## Description

The single-user stories feed at `slash-username` is not visible to logged-out users in development, because the [bullet gem](https://github.com/flyerhzm/bullet) is complaining about an un-used `includes(:commentable)` eager-loading when users are logged-out. If we remove `includes(:commentable)`, however, then bullet complains about insufficient eager-loading when users are logged-_in_. Trying to find a solution that maintains [comment-spam suppression measures](https://github.com/forem/forem/pull/18635) and works in both modes. This ... seems to work?

## Related Tickets & Documents

- Related Issue #18635
- Discovered while working on #19144

## QA Instructions, Screenshots, Recordings

Bullet errors are annoying to troubleshoot, because the page does not render at all — the server appears to have stopped responding. The error is visible in the server log output:

```
 GET /dare_napoleon
 AVOID eager loading detected
   Comment => [:commentable]
   Remove from your query: .includes([:commentable])
 Call stack
   /Users/joshua/forem/forem/app/views/users/_main_feed.html.erb:16:in '_app_views_users__main_feed_html_erb__200249748738175113_324540'
   /Users/joshua/forem/forem/app/views/users/show.html.erb:138:in '_app_views_users_show_html_erb___1460531540532908771_315900'
   /Users/joshua/forem/forem/app/controllers/stories_controller.rb:200:in 'handle_user_index'
   /Users/joshua/forem/forem/app/controllers/stories_controller.rb:110:in 'handle_user_or_organization_or_podcast_or_page_index'
   /Users/joshua/forem/forem/app/controllers/stories_controller.rb:29:in 'index'
```

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [x] No, and this is why: somewhat mysteriously, this does not appear to be covered in test?

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media2.giphy.com/media/1zJaB28JqGw3m/giphy.gif?cid=ecf05e472u0eq5ts4v1zfhjowr0dkjfc14gkjjar571ghbrs&rid=giphy.gif&ct=g)
